### PR TITLE
[Qt] compatibility column the second encounter

### PR DESCRIPTION
**Pull request by Megamouse**
This fixes the crashes that occured on scribams PR.

first two commits are random lol

Let me know what you think

I guess we need to link openssl for it to work on every pc. hnng

### CODE REVIEW

**Comment by JackONeill12** [comment]
I've got errors for all games I have. Like this one:

E Compatibility error: { Error creating SSL context () } | game: { Serial: NPUB30418- Name: From Dust - DEMO- Category: HDD Game }

**Comment by Asinin3** [comment]
Having the same issue here for all my games.

**Comment by danilaml** [comment]
Not sure that this approach (fetching compatibility on each refresh from the web)  is very good.  Dolphin, for example, uses offline ini DB, which also stores best compat options.

So, another approach, without creating such db in the repo,  would probably be to generate something like this in the appveyor when creating an artifact/build. I.e., where could be some compat. db generated from the website stored locally that rpcs3 will use to display compat. ratings. This has the benefit of the DB being a snapshot of compat. at the time of the build creation (so there shouldn't be any worries regarding a user using older build but seeing the higher compat. for a newer version).

**Comment by twdarkeh** [comment]
But most people getting a new build just paste the exe, so there should probably be some way to update the db from wtihin rpcs3.

**Comment by AniLeo** [comment]
Hm can also be done, downloading db from within rpcs3 after downloading perhaps

**Comment by danilaml** [comment]
@twdarkeh they can just paste the db the same way. The db could be even provided as a big json by the rpcs3.net, so rpcs3 would just look for it and parse it with no internet communication. Compat. db is not that important thing and doesn't change so often that'd it warrant an internet request to the rpcs3 web server every time someone launches rpcs3/updates the list. Besides, you'd want it to be cached somewhere anyway in case you play offline/rpcs3 is down,

**Comment by Megamouse** [comment]
yeah that way I can spare all the annoying (and buggy) connection stuff.
I already asked ani if such a thing would be easy to implement into compatibility db.

about SSL. to my knowledge Qt 5.10 will provide openssl 1.1 backend, so we don't have to link it ourself in case we update Qt (hopefully)

**Comment by danilaml** [comment]
@Megamouse what do you mean by `we don't have to link it ourself in case we update Qt`? Qt already dynamically links with openSSL by default. It's jsut that it can't legally ship openSSL dlls/libs along with qt distribution, so you would either ship them yourself (if it's allowed in the target country) or you would tell your users to install it and then your program would just pick it up at runtime.

**Comment by Megamouse** [comment]
hmm . but will it be still necessary then?

**Comment by danilaml** [comment]
@Megamouse to my knowledge - yes.
>Due to import and export restrictions in some parts of the world, we are unable to supply the OpenSSL Toolkit with Qt packages. Developers wishing to use SSL communication in their deployed applications should either ensure that their users have the appropriate libraries installed, or they should consult a suitably qualified legal professional to ensure that applications using code from the OpenSSL project are correctly certified for import and export in relevant regions of the world.

From [here](http://doc.qt.io/qt-5/ssl.html#import-and-export-restrictions)

**Comment by Megamouse** [comment]
Look under QNetwork:

https://wiki.qt.io/New_Features_in_Qt_5.10

this is where I got my information in case you wonder.

doesn't it mean we get the openssl toolkit and said libraries included in qt 5.10?

**Comment by danilaml** [comment]
@Megamouse no, it means that Qt Network will support OpenSSL 1.1.0 as a back-end. It still would have to be provided by the OS/dev.

**Comment by Megamouse** [comment]
ok. sorry for getting that wrong.

so how will that be managed?
will we provide those dlls or at least give a messagebox with links in case they aren't present?
sounds like hell to have nearly every user ask those questions

**Comment by danilaml** [comment]
@Megamouse well, there are at least two alternative solutions that don't require using any crypto/ssl:
1) Store DB locally and use that (it could as well be a json from some of the calls to https://rpcs3.net/compatibility?api=v1 (not sure if there is a way to get the whole cmpat info, or it could be stored in the repo and periodically updated, it's not a feature that affect emulation in anyway anyway (yet)).
2) Allow "unsecure" (i.e. over regular http) access to compat. db.

In case you decided that simple compat. rating is worth the hassle, you'd probably need to adjust the cmake script to copy OpenSSL to the bin dir on windows (appveyor already has OpenSSL installed). The devs would manage to install it themselves (or ignore it).

Or you could just use something like http://doc.qt.io/qt-5/qsslsocket.html#supportsSsl (maybe there is something better) to test whether the openSSL lib was found on the user's PC and prompt him to install it manually or place it near rpcs3 exe if he wants to enable ratings (and disable them otherwise).

**Comment by Megamouse** [comment]
I recreated it to download once on contextmenu click and save to file.
It has some arbitrary value right now and needs a proper url still.

**Comment by joaozin23** [comment]
E Compatibility error: { Could not read database from file: C:/Users/joaoz/Desktop/rpcs3-v0.0.3-2017-10-06-a8593dcd_win64/GuiConfigs/compat_database.dat }
esboco
esboco

**Comment by Megamouse** [comment]
It should be fully functional now except for build warnings and SSL.

**Comment by MelchiorGaspar** [comment]
Would probably be better to do a db text file easy to read or xml or something...
 frequent update from the build bot @ 
https://ci.appveyor.com/project/rpcs3/rpcs3/history
you could probably get the build system to export a copy from the compatibility db online..

then lol you wouldn't need OpenSSL or networking for it.. no real point to do it that way because
the updated compatibility would only come with new updates to the binary anyway..

only one good reason to have a connection to grab it live like... and that is if it would compare 
current version to the version online  and local disc images... and let the user know if there if their game would get BETTER results with an update to that version or newer... O_o X_x

**Comment by Megamouse** [comment]
but most people only copy over the exe.
that's why the kuroi stylesheet looks funky for some people still.
They would never get the newest and even if, if they aren't careful and copy over the list of some old build it is outdated.

**Comment by danilaml** [comment]
@Megamouse it's their issue then since, as you said, rpcs3 already may function wonky in such case. At worst they'll just have outdated db, no big deal.

**Comment by Asinin3** [comment]
What about just downloading a most recent version of the DB through a UI button? Then if the tested date is after the date the build was created it greys out? Just because the user has an old build doesn't mean a game listed as not compatible won't work. I'm constantly finding games that need to be moved category, though this mostly applies to smaller titles.

File: rpcs3/CMakeLists.txt [hunk 1]
```diff
@@ -6,17 +6,17 @@ set(CMAKE_CXX_STANDARD 14)
 include(CheckCXXCompilerFlag)
 
 # Qt section
-find_package(Qt5 5.7 COMPONENTS Widgets)
+find_package(Qt5 5.7 COMPONENTS Widgets Network)
 if(WIN32)
 	find_package(Qt5 5.7 COMPONENTS WinExtras REQUIRED)
-	set(RPCS3_QT_LIBS Qt5::Widgets Qt5::WinExtras)
+    set(RPCS3_QT_LIBS Qt5::Widgets Qt5::WinExtras Qt5::Network)
```
Comment (scribam) [inline rpcs3/CMakeLists.txt]: Tab instead of spaces?

Comment (Megamouse) [reply rpcs3/CMakeLists.txt]: 😱

**Pull request merged**